server/questions.js

```diff
@@ -196,7 +202,35 @@ exports.checkAnswer = function(questionId, user, answer, callback) {
 			return callback('error', null);
 		}
 
-		var value = (answer === question.answer);
+		var value = null;
+
+		if (question.type === common.questionTypes.MATCHING.value && answer) {
+		    const ansLeftSide = answer[0];
```

Comment:
Good idea, I think we should keep the matching question checking part in its own function and you can reference it in here by using cases as we do for other types. That way this function is clean and we can add more checking for other question types using that approach

Reply:
sounds good